reproject/common.py

```diff
@@ -16,6 +16,7 @@
 
 @delayed(pure=True)
 def as_delayed_memmap_path(array, tmp_dir):
+    tmp_dir = tempfile.mkdtemp()  # FIXME
```

Comment:
I'm not sure why I didn't catch this before but I ran into an issue where the temporary directory no longer exists by the time the array is computed out of dask when using ``return_type='dask'``.

Reply:
This is a temporary fix for this PR but I need to figure out a proper solution. It is a problem in ``main``.